lib/deploy.rb

```diff
@@ -22,24 +24,42 @@ def run
       check_for_unstaged_changes!
       check_for_changelog!
       set_aws_region!
+      fetch_eb
       verify_configuration!
 
-      @name, @app_bucket = select_app
-      @platform          = detect_platform
+      @name       = deployment_target
+      @platform   = detect_platform
 
       request_confirmation!
       synchronize_repo!
       deploy!
```

Comment:
Except that the Github view is hiding that there is a run method already, that encompasses all of that.